tools/data/textdet/ilst_converter.py

```diff
@@ -0,0 +1,206 @@
+
+    if nproc > 1:
+        images = mmcv.track_parallel_progress(
+            load_img_info, files, nproc=nproc)
+    else:
+        images = mmcv.track_progress(load_img_info, files)
+
+    return images
+
+
+def load_img_info(files):
+    """Load the information of one image.
+
+    Args:
+        files (tuple): The tuple of (img_file, groundtruth_file)
+
+    Returns:
+        img_info (dict): The dict of the img and annotation information
+    """
+    assert isinstance(files, tuple)
+
+    img_file, gt_file = files
+    assert osp.basename(gt_file).split('.')[0] == osp.basename(img_file).split(
+        '.')[0]
+    # read imgs while ignoring orientations
+    img = mmcv.imread(img_file, 'unchanged')
+
+    img_info = dict(
+        file_name=osp.join(osp.basename(img_file)),
+        height=img.shape[0],
+        width=img.shape[1],
+        segm_file=osp.join(osp.basename(gt_file)))
+
+    if osp.splitext(gt_file)[1] == '.xml':
+        img_info = load_xml_info(gt_file, img_info)
+    else:
+        raise NotImplementedError
+
+    return img_info
+
+
+def load_xml_info(gt_file, img_info):
+    """Collect the annotation information.
+
+    Args:
+        gt_file (str): The path to ground-truth
+        img_info (dict): The dict of the img and annotation information
+
+    Returns:
+        img_info (dict): The dict of the img and annotation information
+    """
+    obj = ET.parse(gt_file)
+    root = obj.getroot()
+    anno_info = []
+    for object in root.iter('object'):
+        word = object.find('name').text
+        iscrowd = 1 if len(word) == 0 else 0
+        x1 = int(object.find('bndbox').find('xmin').text)
+        y1 = int(object.find('bndbox').find('ymin').text)
+        x2 = int(object.find('bndbox').find('xmax').text)
+        y2 = int(object.find('bndbox').find('ymax').text)
+
+        x = max(0, min(x1, x2))
+        y = max(0, min(y1, y2))
+        w, h = abs(x2 - x1), abs(y2 - y1)
+        bbox = [x1, y1, w, h]
+        segmentation = [x, y, x + w, y, x + w, y + h, x, y + h]
+        anno = dict(
+            iscrowd=iscrowd,
+            category_id=1,
+            bbox=bbox,
+            area=w * h,
+            segmentation=[segmentation])
+        anno_info.append(anno)
+
+    img_info.update(anno_info=anno_info)
+
+    return img_info
+
+
+def judge_latin(word):
+    for char in word:
+        if char not in string.ascii_letters:
+            return False
+    return True
+
+
+def split_train_test_list(full_list, test_ratio, shuffle=True):
+    """Split list by test_ratio
+
+    Args:
+        full_list (list): List to be splited
+        test_ratio (float): Ratio for test set
+        shuffle (bool): Whether to random sample test set
+
+    return:
+        list(list, list): train_list and test_list
+    """
+
+    n_total = len(full_list)
+    offset = int(n_total * test_ratio)
+    if n_total == 0 or offset < 1:
+        return [], full_list
+    if shuffle:
+        random.shuffle(full_list)
```

Comment:
We may do not need a random shuffle. Since the results will be different for different users, or when you run the data converter at the next time. But we do need to ensure that the images in different folders will be selected as both training and test.

Reply:
Or you can set the shuffle as 'False' as default

Reply:
Agreed. Random shuffle is usually a part of the data loading phase, where users can customize a random seed to control the randomness. This option breaks the assumption and harms reproducibility.

Reply:
Get it. I think I should just remove the shuffle option